server/models/procedures/invoicing.sql

```diff
@@ -980,11 +984,13 @@ CREATE PROCEDURE UnbalancedInvoicePaymentsTable(
   CREATE TEMPORARY TABLE `unbalanced_invoices` AS (
     SELECT BUID(ivc.uuid) as invoice_uuid , em.text AS debtorReference, debtor.text AS debtorName,
       BUID(debtor.uuid) as debtorUuid,
-      balances.debit_equiv AS debit,
-      balances.credit_equiv AS credit, iv.date AS creation_date, balances.balance,
+      (balances.debit_equiv * IFNULL(GetExchangeRate(_enterpriseId, currencyId, dateTo), 1)) AS debit,
```

Comment:
It looks like you can speed this up a lot by moving the `GetExchangeRate(_enterpriseId, currencyId, dateTo)` out of the SQL query.  Like this:

```sql
DECLARE exchangeRate DOUBLE;

SET exchangeRate = (SELECT IFNULL(GetExchangeRate(_enterpriseId, currencyId, dateTo), 1));

  CREATE TEMPORARY TABLE `unbalanced_invoices` AS (
    SELECT BUID(ivc.uuid) as invoice_uuid , em.text AS debtorReference, debtor.text AS debtorName,
      BUID(debtor.uuid) as debtorUuid,
      (balances.debit_equiv * exchangeRate) AS debit,
       /* etc ... */
```

This has the advantage of speeding up the query since the `GetExchangeRate()` function doesn't depend on the data in the table.  Instead of calling it 3X per row, we only call it a single time.